nemo/collections/llm/deploy/fastapi_interface_to_pytriton.py

```diff
@@ -0,0 +1,192 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+
+import numpy as np
+import requests
+from fastapi import FastAPI, HTTPException
+from jinja2 import Template
+from pydantic import BaseModel
+from pydantic_settings import BaseSettings
+
+from nemo.deploy.nlp import NemoQueryLLMPyTorch
+from nemo.utils import logging
+
+
+class TritonSettings(BaseSettings):
+    _triton_service_port: int
+    _triton_service_ip: str
+
+    def __init__(self):
+        super(TritonSettings, self).__init__()
+        try:
+            self._triton_service_port = int(os.environ.get('TRITON_PORT', 8000))
+            self._triton_service_ip = os.environ.get('TRITON_HTTP_ADDRESS', '0.0.0.0')
+        except Exception as error:
+            logging.error("An exception occurred trying to retrieve set args in TritonSettings class. Error:", error)
+            return
+
+    @property
+    def triton_service_port(self):
+        return self._triton_service_port
+
+    @property
+    def triton_service_ip(self):
+        return self._triton_service_ip
+
+
+app = FastAPI()
+triton_settings = TritonSettings()
+
+
+class CompletionRequest(BaseModel):
+    model: str
+    prompt: str = 'hello'
+    messages: list[dict] = [{}]
+    max_tokens: int = 512
+    temperature: float = 1.0
+    top_p: float = 0.0
+    top_k: int = 1
+    logprobs: int = 1
```

Comment:
Maybe we can set the default to 0? I'm not sure if this is in the official API spec, but lm-eval-harness assumes that 0 is the default and 1 needs to be passed ([here](https://github.com/EleutherAI/lm-evaluation-harness/blob/1ba35e623b9bd9ca48df926f1a028043e159a6f2/lm_eval/models/openai_completions.py#L59))